### PR TITLE
Bump: New minimum requirement for ansible-core 2.16

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Change Summary

New minimum requirement for ansible-core 2.16

## Component(s) name

Python requirements

## Proposed changes

- [x] requirements
  - [x] pyproject.toml
  - [x] meta/runtime.yml
  - [x] requirements-dev.txt
- [x] pre-commit.yaml configuration with new min and max values
- [x] GitHub worflows/ pull-request-management.yml to test new minimum version
- [x] 5.2.0 release notes

## How to test

CI Should pass!

### CODE REVIEW

**Comment by alexeygorbunov** [comment]
This file is not exposed as a part of the documentation but probably needs a bump as well?

https://github.com/carlbuchmann/avd/blob/a47cb5795dad933dd9d96397df74636565155b2a/ansible_collections/arista/avd/README.md?plain=1#L25

File: .github/workflows/pull-request-management.yml [hunk 1]
```diff
@@ -216,10 +216,10 @@ jobs:
         # Also test minimum ansible version for one scenario.
         include:
           - avd_scenario: 'eos_designs_unit_tests'
-            ansible_version: 'ansible-core==2.15.0'
+            ansible_version: 'ansible-core==2.16.0'
             pip_requirements: 'tmp-requirements-minimum.txt'
           - avd_scenario: 'eos_designs_unit_tests'
-            ansible_version: 'ansible-core>=2.15.0,<2.16.0 --upgrade'
+            ansible_version: 'ansible-core>=2.16.0,<2.17.0 --upgrade'
```
Comment (ClausHolbechArista) [inline .github/workflows/pull-request-management.yml]: We don't need this one, since it would just test 2.16.<latest>. We can remove this variant.

Comment (carlbuchmann) [reply .github/workflows/pull-request-management.yml]: updated and removed workflow.

**Comment by ClausHolbechArista** [comment]
> This file is not exposed as a part of the documentation but probably needs a bump as well?
> 
> https://github.com/carlbuchmann/avd/blob/a47cb5795dad933dd9d96397df74636565155b2a/ansible_collections/arista/avd/README.md?plain=1#L25

fixed. But that readme should be rewamped to basically just point to the documentation site.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]
